File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> pairIdx = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            if (pairIdx.containsKey(target - nums[i])) {
                return new int[]{i, pairIdx.get(target - nums[i])};
            }
            pairIdx.put(nums[i], i);
        }

        return new int[]{};
    }
}

File: 0001-two-sum/0001-two-sum.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val pairIdx = mutableMapOf<Int, Int>()

        for ((i, num) in nums.withIndex()) {
            if (pairIdx.containsKey(target - num)) {
                return intArrayOf(pairIdx[target - num] ?: 0, i)
            }
            pairIdx[num] = i
        }

        return intArrayOf()
    }
}